### PR TITLE
tests/samples: use integration_platforms more where it makes sense

### PR DESCRIPTION
**Pull request by nashif**
Use integration platforms to limit churn and build time in PR CI .

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

### CODE REVIEW

File: samples/subsys/display/lvgl/sample.yaml [hunk 1]
```diff
@@ -17,3 +17,5 @@ tests:
       - gui
     modules:
       - lvgl
+    integration_platforms:
+      - mimxrt1064_evk
```
Comment (gmarull) [inline samples/subsys/display/lvgl/sample.yaml:21]: note: native_posix can be an option as well (with SDL display)

Comment (nashif) [reply samples/subsys/display/lvgl/sample.yaml:21]: yeah, tried this, but this requires 32bit SDL2 on the host and not everyone will have that.

Comment (gmarull) [reply samples/subsys/display/lvgl/sample.yaml:21]: so maybe `native_posix_64`? `libsdl2-dev` seems to be a Zephyr requirement: https://docs.zephyrproject.org/latest/develop/getting_started/index.html, but... it probably does not belong to getting started...

**Review by cfriedt — APPROVED** [review]
LGTM

**Review by jfischer-no — APPROVED** [review]

**Pull request merged**